### ISSUE TITLE
Add pycln example

### ISSUE THREAD

**Issue by RoyalTS** [nbQA-dev/nbQA#634]
Seems like [`pycln`](https://hadialqattan.github.io/pycln/) would be a nice addition.

**Comment by MarcoGorelli** [comment]
If it works, looks good to me! Do you want to make a PR?

**Comment by RoyalTS** [comment]
This kinda sorta works:

```yaml
- repo: https://github.com/hadialqattan/pycln
  rev: 0.0.4
  hooks:
  - id: pycln
    args: [--config=pyproject.toml]

- repo: https://github.com/nbQA-dev/nbQA
  rev: 1.0.0
  hooks:
  - id: nbqa
    args: [pycln]
    name: nbqa-pycln
    alias: nbqa-pycln
    additional_dependencies: [pycln]
```
    
but doesn't respect `pyproject.toml` and I'm not entirely clear on how I would pass additional arguments to `pycln` in the [custom pre-commit hook specification](https://nbqa.readthedocs.io/en/latest/pre-commit.html#custom-hooks).

**Comment by RoyalTS** [comment]
... though I guess I could pass them in `pyproject.toml` under `[tool.nbqa.addopts]`

**Comment by MarcoGorelli** [comment]
> but doesn't respect pyproject.toml

:thinking: if that's the case, then that would be a bug - could you please share your pyproject.toml file and a sample notebook which reproduces the issue?

**Comment by RoyalTS** [comment]
We may have misunderstood each other. `pycln` needs to be passed the `--config=pyproject.toml` command line option. What naively I did was

```yaml
- repo: https://github.com/nbQA-dev/nbQA
  rev: 1.0.0
  hooks:
  - id: nbqa
    args: [pycln --config=pyproject.toml]
    name: nbqa-pycln
    alias: nbqa-pycln
    additional_dependencies: [pycln]
```

But that results in

```sh
ModuleNotFoundError: Command `pycln --config=pyproject.toml` not found by nbqa.

Please make sure you have it installed in the same Python environment as nbqa. See
e.g. https://realpython.com/python-virtual-environments-a-primer/ for how to set up
a virtual environment in Python, and run:

    `python -m pip install pycln --config=pyproject.toml`.
```

What _does_ work is to leave off the command line option in the above and specify the following in pyproject.toml:

```toml
[tool.nbqa.addopts]
pycln = ["--config=pyproject.toml"]
```

**Comment by MarcoGorelli** [comment]
Ah, I see, thanks for explaining!

I think the docs need to be updated, it would be better to do:
```
- repo: https://github.com/nbQA-dev/nbQA
  rev: 1.0.0
  hooks:
    - id: nbqa
      args: ["--config=pyproject.toml"]
      name: nbqa-pycln
      alias: nbqa-pycln
      entry: nbqa pycln
      additional_dependencies: [pycln]
```

does that work for you?

**Comment by MarcoGorelli** [comment]
As for adding a built-in hook for `pycln` - I'm hesitant to "officially" add anything which removes code, as it it might be that something is imported but is only used [in a cell which nbQA can't process](https://nbqa.readthedocs.io/en/latest/known-limitations.html)

**Comment by MarcoGorelli** [comment]
fixed the docs in 9bad17492b2bff9d7edbeb640993fd1629673032 , please do let me know if this doesn't work and I'll reopen

**Comment by RoyalTS** [comment]
No, that's brilliant. Thank you!
